harness/determined/cli/workspace.py

```diff
@@ -171,21 +170,36 @@ def list_workspace_namespace_bindings(args: argparse.Namespace) -> None:
 
 def set_workspace_namespace_binding(args: argparse.Namespace) -> None:
     sess = cli.setup_session(args)
-    if args.cluster_name and not args.namespace:
-        raise api.errors.BadRequestException("must provide --namespace NAMESPACE")
+    if args.cluster_name and not (
+        args.namespace or args.auto_create_namespace or args.auto_create_namespace_all_clusters
+    ):
+        raise api.errors.BadRequestException(
+            "must provide --namespace NAMESPACE, "
+            + "--auto-create-namespace, or --auto-create-namespace-all-clusters"
+        )
+
     w = api.workspace_by_name(sess, args.workspace_name)
     content = bindings.v1SetWorkspaceNamespaceBindingsRequest(workspaceId=w.id)
     cluster_name = "" if not args.cluster_name else args.cluster_name
-    content.clusterNamespacePairs = {cluster_name: args.namespace}
+    requested_namespace = args.namespace or ""
+    namespace_meta = bindings.v1WorkspaceNamespaceMeta(
+        namespace=requested_namespace,
+        autoCreateNamespace=args.auto_create_namespace,
+        autoCreateAllNamespaces=args.auto_create_all_namespaces,
+    )
+    content.clusterNamespaceMeta = {cluster_name: namespace_meta}
 
     resp = bindings.post_SetWorkspaceNamespaceBindings(sess, body=content, workspaceId=w.id)
     for cluster_name in resp.namespaceBindings:
         cluster_details = ""
-        if args.cluster_name:
+        if cluster_name:
```

Comment:
Not blocking: when `cluster_name` might end up being an empty string?

Reply:
When the user is running singleRM determined (where a specifying cluster name in master config isn't required)